MyMovies/Model Controllers/MovieController.swift

```diff
@@ -61,4 +63,34 @@ class MovieController {
             }
         }.resume()
     }
+    
+    
+    func fbGetAllMovies(completion: @escaping  ((Error?) -> Void) = { _ in}) {
+        
+        let url = firebaseUrl.appendingPathExtension("json")
+        
+        URLSession.shared.dataTask(with: url) { (data, _, error) in
+            if let error = error {
+                completion(error)
+                return
+            }
+            
+            guard let data = data else {
+                completion(error)
+                return
+            }
+            
+            let decoder = JSONDecoder()
+            
+            do {
+                let movies = try decoder.decode([MovieRepresentation].self, from: data)
+                let moc = CoreDataStack.shared.container.newBackgroundContext()
+                
+                completion(nil)
```

Comment:
So think about how your data from firebase will be coming in. I added a snippet below I like to use because it prints the json very clearly which helps me with decoding issues.
```suggestion
            do {
                let json = try! JSONSerialization.jsonObject(with: data, options: [])
                print(json)
                let movies = try decoder.decode([MovieRepresentation].self, from: data)
                let moc = CoreDataStack.shared.container.newBackgroundContext()
                
                completion(nil)
```